### PR TITLE
Minor build and CI fixes

### LINKED ISSUE #1593: Minor patch for gcc 12
As it typically happens, some headers are no longer "free" with major gcc upgrades. I needed to add this in order to build with gcc 12:

```
Index: OpenColorIO-2.1.1/src/OpenColorIO/FileRules.cpp
===================================================================
--- OpenColorIO-2.1.1.orig/src/OpenColorIO/FileRules.cpp
+++ OpenColorIO-2.1.1/src/OpenColorIO/FileRules.cpp
@@ -3,6 +3,7 @@
 
 #include <algorithm>
 #include <cctype>
+#include <cstring>
 #include <map>
 #include <regex>
 #include <sstream>
```

Pretty minor otherwise I would have done a PR.

### PR DESCRIPTION
**Pull request by remia**
#### Fix the analysis workflow

The install location for built dependencies in the analysis workflow is now explicit in Linux and macOS (Windows already use a custom location). This is because OpenImageIO now started to override the install location to point in the build tree when none is specified, instead of using standard Unix paths.

Currently failing on OpenEXR Windows build, see related issue https://github.com/AcademySoftwareFoundation/openexr/issues/1236

#### Improve documentation for OFX build #1607

### CODE REVIEW

**Comment by remia** [comment]
Note that this PR is not complete yet and will probably include some fix related to #1581.

File: tests/cpu/Config_tests.cpp [hunk 1]
```diff
@@ -8676,3 +8676,63 @@ OCIO_ADD_TEST(Config, look_is_noop)
         OCIO_CHECK_ASSERT(proc->isNoOp());
     }
 }
+
+OCIO_ADD_TEST(Config, look_fallback)
+{
+    // Test that the look fallback syntax works for look with missing file.
```
Comment (hodoulp) [inline tests/cpu/Config_tests.cpp:8682]: Please add a comment to explain what is the meaning of a look fallback `missing_file_look | ` in the unit test. This notation can be used for fallback but it mainly highlights that there are two choices (i.e. `missing_file_look` and `none`) and the precessing takes the first successful.

Comment (remia) [reply tests/cpu/Config_tests.cpp:8682]: Good point, comment added.

**Comment by hodoulp** [comment]
@remia What is the status of this pull request?

File: tests/cpu/Config_tests.cpp [hunk 2]
```diff
@@ -8680,6 +8680,10 @@ OCIO_ADD_TEST(Config, look_is_noop)
 OCIO_ADD_TEST(Config, look_fallback)
 {
     // Test that the look fallback syntax works for look with missing file.
+    // The fallback syntax allow to specify looks to try in order and use the
+    // first valid one. When the syntax "my_look | " is used, the fallback
+    // is empty and there will be no look applied (no-op) if my_look in invalid.
```
Comment (hodoulp) [inline tests/cpu/Config_tests.cpp:8685]: typo `if my_look in invalid` to `if my_look is invalid`

File: tests/cpu/Config_tests.cpp [hunk 3]
```diff
@@ -8680,6 +8680,10 @@ OCIO_ADD_TEST(Config, look_is_noop)
...
+    // This may happen if my_look rely on a missing environment variable.
```
Comment (hodoulp) [inline tests/cpu/Config_tests.cpp:8686]: typo `rely` to `relies`

**Comment by remia** [comment]
> @remia What is the status of this pull request?

This PR is ready to merge now from my point of view @hodoulp.

**Review by doug-walker — APPROVED** [review]
Thanks Remi!

There seems to be an analysis build failure on Windows related to OpenEXRCore, but I didn't try to figure out why it's happening on one Windows build but not another.  Any thoughts on what went wrong?

**Comment by remia** [comment]
> There seems to be an analysis build failure on Windows related to OpenEXRCore, but I didn't try to figure out why it's happening on one Windows build but not another. Any thoughts on what went wrong?

Hi @doug-walker, this is the issue I mentioned in the PR description, OpenEXR doesn't compile on Windows MSVC 2022 currently, while the issue has now been fixed upstream, it's not released yet. I expect the analysis build will be fixed when that is the case. We shouldn't wait for this however.

**Pull request merged**